x-pack/metricbeat/module/aws/cloudwatch/cloudwatch_test.go

```diff
@@ -1406,7 +1420,7 @@ func TestCreateEventsWithTagsFilter(t *testing.T) {
 		},
 	}
 
-	events, err = m.createEvents(mockCloudwatchSvc, mockTaggingSvc, listMetricWithStatsTotal, resourceTypeTagFilters, regionName, startTime, endTime)
+	events, err = m.createEvents(mockCloudwatchSvc, mockTaggingSvc, listMetricWithStatsTotal, resourceTypeTagFilters, infoAPImap, regionName, startTime, endTime)
```

Comment:
I wonder if we can store infoAPImap result into resourceTypeTagFilters since it is collecting tags but just with a diff api. This way, we dont have to change the createEvents function input to have a special parameter for apigatewayapi.

Reply:
The infoAPImap has [Name_of_IP : ID_of_API] not actual tags. 
The diff:  resourceTypeTagFilters map[string][]aws.Tag and  infoAPImap map[string]string,
(I will add it somewhere in a note to make clear)

TBH we can create a new createEventsforAPIGW that will have the new infoAPImap but too many same code lines just for one argument. WDYT?

Reply:
Yeah lets keep it as is then, thanks!!